core/dictgen/src/rootcling_impl.cxx

```diff
@@ -4571,7 +4570,12 @@ int RootClingMain(int argc,
    }
 
    // Now generate a second stream for the split dictionary if it is necessary
-   std::ostream *splitDictStreamPtr = gOptSplit ? CreateStreamPtrForSplitDict(dictpathname, tmpCatalog) : dictStreamPtr;
+   std::unique_ptr<std::ostream, void (*)(std::ostream *)> splitDictStreamPtr{nullptr, nullptr};
+   if (gOptSplit) {
+      splitDictStreamPtr = CreateStreamPtrForSplitDict(gOptDictionaryFileName.getValue(), tmpCatalog);
+   } else {
+      splitDictStreamPtr = {dictStreamPtr, [](std::ostream *) {}};
```

Comment:
You may exploit the fact the the `unique_ptr::get_deleter()` returns a non-const.

So, you define your regular unique_ptr and in the else branch you set the deleter to some nop routine. I am not sure if that would make it better but would make the ugliness contained in a single statement of a single branch.